### PR TITLE
Add React language track with agents, skills, rules, and commands

### PR DESCRIPTION
**Pull request by Alexsky347**
## What Changed

  Add full React language track mirroring existing tracks (Go, Rust, Kotlin, etc.):
  
  - **Rules** (`rules/react/`): `coding-style`, `hooks`, `patterns`, `security`, `testing`
  - **Skills**: `react-patterns`, `react-testing`, `react-performance` (SKILL.md each)
  - **Agents**: `react-reviewer` (component patterns, hook rules, WCAG AA, perf budgets, security) and `react-build-resolver` (Vite/Next/CRA build + type error fixes). Kiro JSON variants under `.kiro/agents/`.
  - **Commands**: `/react-review`, `/react-build`, `/react-test`
  - **Wiring**: registered in `agent.yaml`, `config/project-stack-mappings.json`, `docs/COMMAND-REGISTRY.json`, `manifests/install-modules.json`; cross-references added in `CLAUDE.md`, `AGENTS.md`, `README.md`,
  `README.zh-CN.md`, `docs/zh-CN/*`, `agents/typescript-reviewer.md`
  - Catalog counts synced: 62 agents / 78 commands / 235 skills

  ## Why This Change

  Repo had TS/JS coverage but no React-specific lane. React work hits patterns the generic TS reviewer misses: hook rules, Server/Client component boundaries, render-perf budgets, accessibility, JSX-specific XSS surface. New
  track gives users `/react-review`, `/react-build`, `/react-test` parity with other language tracks and avoids overloading `typescript-reviewer` with framework concerns.

  ## Testing Done

  - [x] Manual testing completed — invoked `/react-review`, `/react-build`, `/react-test` and verified registry resolution
  - [x] Automated tests pass locally (`node tests/run-all.js`)
  - [x] Edge cases considered and tested — confirmed `react-reviewer` does not duplicate `typescript-reviewer` scope; added cross-reference note in `agents/typescript-reviewer.md`

  ## Type of Change

  - [x] `feat:` New feature

  ## Security & Quality Checklist

  - [x] No secrets or API keys committed
  - [x] JSON files validate cleanly (`.kiro/agents/*.json`, `docs/COMMAND-REGISTRY.json`, `config/project-stack-mappings.json`)
  - [ ] Shell scripts pass shellcheck — N/A (no shell scripts changed)
  - [x] Pre-commit hooks pass locally
  - [x] No sensitive data exposed in logs or output
  - [x] Follows conventional commits format (`feat:`, `chore:`)

  ## Documentation

  - [x] Updated relevant documentation (`CLAUDE.md`, `AGENTS.md`, `README.md`, `README.zh-CN.md`, `docs/zh-CN/AGENTS.md`, `docs/zh-CN/README.md`, `docs/COMMAND-REGISTRY.json`)
  - [x] Added comments for complex logic — N/A, markdown/config only
  - [x] README updated — React track listed in language table

---
## Summary by cubic
Adds a full React language track with rules, skills, agents, and commands for React-focused review, build fixes, and TDD. Syncs catalog counts/docs to 63 agents / 79 commands / 249 skills and finalizes the command registry by adding `golang-testing` and removing a stale `kotlin-patterns` entry.

- **New Features**
  - React rules: coding-style, hooks, patterns, security, testing (auto-activated via path globs).
  - Skills: `react-patterns`, `react-testing`, `react-performance`.
  - Agents: `react-reviewer` and `react-build-resolver` (Vite/webpack/Next.js/CRA/Parcel/esbuild/Bun/Rsbuild).
  - Commands: `/react-review`, `/react-build`, `/react-test`; uses `eslint-plugin-react-hooks`, `eslint-plugin-jsx-a11y`, `Vitest` or `Jest` when detected.
  - Wiring: added to `agent.yaml`, stack mappings (rules include "react"; skills add accessibility), command registry, manifests; TS reviewer now defers React checks to the React reviewer.

- **Bug Fixes**
  - Removed unsafe file `undefined/.claude/session-aliases.json` containing `__proto__`.
  - Command registry: regenerated to add `golang-testing` and drop stale `kotlin-patterns`; CI check green.
  - `react-build-resolver`: sturdier bundler detection; version-agnostic React/ReactDOM pair-upgrade guidance; guard typecheck with `test -f tsconfig.json && npx --no-install tsc --noEmit -p tsconfig.json` (mirrored in `.kiro` JSON).
  - `/react-review`: guard `tsc --noEmit -p tsconfig.json` for JS-only repos.
  - React security rules: corrected `javascript:` URL note and CRA env-var exposure; other refinements to `rules/react/security.md`.
  - `react-testing` skill: persist a single `QueryClient` and always `mockRestore()` `console.error`.
  - Docs: fixed nested code fences in `/react-test`, `/react-build`, `/react-review`; clarified README commands row as “79 commands (legacy shims)” and relaxed the catalog validator regex to allow an optional parenthetical after the command count.
  - `.gitignore`: ignore `.claude/settings.local.json` to keep local Claude settings out of the PR.

<sup>Written for commit c68e3022ca94df0c990482b7371f62a88230ff2c. Summary will update on new commits.</sup>

<a href="https://cubic.dev/pr/affaan-m/ECC/pull/2024?utm_source=github">Review in cubic</a>

## Summary by CodeRabbit

* **New Features**
  * Added React-focused agents and slash commands: /react-review, /react-build, and /react-test to review, diagnose, and incrementally fix React code, builds, and tests.
  * Added three React skills: react-patterns, react-performance, and react-testing.
  * Added comprehensive React guidance for coding style, hooks, patterns, security, testing, and performance.

* **Documentation**
  * Updated public inventory counts to 62 agents, 235 skills, and 78 commands; refreshed docs and registries.

[Review Change Stack](https://app.coderabbit.ai/change-stack/affaan-m/ECC/pull/2024?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: commands/react-test.md [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+```text
+User: /react-test I need a SearchInput component with debounced search
+
+Agent:
+# TDD Session: SearchInput
+
+## Step 1: Define Signature
+
+```tsx
+// src/components/SearchInput.tsx
```
[the remark this answers is not in the document]
Comment (Alexsky347) [reply commands/react-test.md]: Fixed in c164c471 — the .kiro JSON variant now mirrors the conditional, version-agnostic guidance applied to the .md companion 
  in de135f61

File: undefined/.claude/session-aliases.json [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+      "createdAt": "2026-05-20T09:16:26.009Z",
+      "title": "Prototype Pollution Attempt"
+    },
+    "normal": {
+      "sessionPath": "/normal/path",
+      "createdAt": "2026-05-20T09:16:26.009Z",
+      "title": "Normal Alias"
+    }
+  },
+  "metadata": { "totalCount": 2, "lastUpdated": "2026-05-20T09:16:26.009Z" }
```
[the remark this answers is not in the document]
Comment (Alexsky347) [reply undefined/.claude/session-aliases.json]: pullrequestreview-4327246111 referencing de135f61

File: commands/react-build.md [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+## Initial Diagnostics
+
+```bash
+$ npm run build
+> tsc -b && vite build
+
+src/components/UserCard.tsx:1:8 - error TS6133: 'React' is declared but its value is never read.
+src/components/Modal.tsx:12:15 - error TS7016: Could not find a declaration file for module 'react-portal'.
+src/pages/Home.tsx:42:5 - error: 'useState' is not defined
+```
```
[the remark this answers is not in the document]
Comment (Alexsky347) [reply commands/react-build.md]: Fixed in a5091877 — outer  text` fence bumped to 4 backticks ` ````text`, matching the closing fence at line 143, so
  the inner ` bash/ ```tsx blocks no longer terminate it prematurely

File: commands/react-review.md [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+[CRITICAL] Unsanitized dangerouslySetInnerHTML
+File: src/components/UserCard.tsx:42
+Issue: User-controlled bio rendered as raw HTML.
+Why: XSS via stored script tags in user input.
+Fix: Sanitize with DOMPurify or render as text:
+```tsx
+import DOMPurify from "isomorphic-dompurify";
+<div dangerouslySetInnerHTML={{ __html: DOMPurify.sanitize(user.bio) }} />
+```
+
```
[the remark this answers is not in the document]
Comment (Alexsky347) [reply commands/react-review.md:127]: Fixed in 37d8569a — outer  ```textbumped to ````text, closing fence at line 148 matched

File: README.md [hunk 5]
```diff
@@ -1423,9 +1423,9 @@ The configuration is automatically detected from `.opencode/opencode.json`.
 
 | Feature | Claude Code | OpenCode | Status |
 |---------|-------------|----------|--------|
-| Agents | PASS: 60 agents | PASS: 12 agents | **Claude Code leads** |
-| Commands | PASS: 75 commands | PASS: 35 commands | **Claude Code leads** |
-| Skills | PASS: 232 skills | PASS: 37 skills | **Claude Code leads** |
+| Agents | PASS: 62 agents | PASS: 12 agents | **Claude Code leads** |
+| Commands | PASS: 78 commands | PASS: 35 commands | **Claude Code leads** |
+| Skills | PASS: 235 skills | PASS: 37 skills | **Claude Code leads** |
```
[the remark this answers is not in the document]
Comment (Alexsky347) [reply README.md]: Fixed in feaebeb9 — comparison-table cell now reads 'PASS: 78 commands (legacy shims)', matching the install-section and quick-start
  phrasing. Catalog validator regex widened to tolerate the optional parenthetical so existing translations still pass

File: docs/COMMAND-REGISTRY.json [hunk 6]
```diff
@@ -1,6 +1,6 @@
 {
   "schemaVersion": 1,
-  "totalCommands": 75,
+  "totalCommands": 78,
```
[the remark this answers is not in the document]
Comment (Alexsky347) [reply docs/COMMAND-REGISTRY.json]: Verified — current registry has totalCommands=78, byType sum=78, commands array length=78. All three aggregates already match. No change needed

File: .kiro/agents/react-build-resolver.json [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
...s or a version mismatch with @types/react.\n# Upgrade react and react-dom as a pair (matching the major already in use) — never independently.\n# Replace <major> with the project React major (17 / 18 / 19); jumping majors is a separate, deliberate change.\n# npm i react@^<major> react-dom@^<major>\n```\n\n## Key Principles\n\n- Surgical fixes only — don't refactor\n- Never disable type-checking or lint rules to make it green\n- Never add `// @ts-ignore` without an inline explanation and a TODO\n- Always re-run the build after each fix — do not stack changes\n- Fix root cause over suppressing symptoms\n- If the error indicates a real architectural problem, stop and report\n\n## Stop Conditions\n\n- Same error persists after 3 fix attempts\n- Fix introduces more errors than it resolves\n- Error requires architectural changes beyond build resolution\n- Bundler version no longer supports the installed React major\n\n## Output Format\n\n```text\n[FIXED] src/components/UserCard.tsx\nError: 'React' is not defined\nFix: tsconfig.json -> set \"jsx\": \"react-jsx\"; removed obsolete import\nRemaining errors: 2\n```\n\nFinal: `Build Status: SUCCESS | Errors Fixed: N | Files Modified: <list>`"
```
[the remark this answers is not in the document]
Comment (Alexsky347) [reply .kiro/agents/react-build-resolver.json]: Fixed in b3e3c989 — guarded with test -f tsconfig.json && npx --yes tsc --noEmit -p tsconfig.json in both agents/react-build-resolver.md and 
  the .kiro JSON prompt mirror

File: agents/react-build-resolver.md [hunk 8]
```diff
@@ -0,0 +1,212 @@
...
+# Run the project's build script first — respect what's configured
+npm run build --if-present
+pnpm build 2>/dev/null
+yarn build 2>/dev/null
+bun run build 2>/dev/null
+
+# Typecheck independently of the bundler — only when TypeScript is configured
+# (skips cleanly for JavaScript-only projects)
+npm run typecheck --if-present
+test -f tsconfig.json && npx --yes tsc --noEmit -p tsconfig.json
```
[the remark this answers is not in the document]
Comment (Alexsky347) [reply agents/react-build-resolver.md]: Fixed in c3464843 — switched to npx --no-install tsc in both agents/react-build-resolver.md and the .kiro JSON mirror so the diagnostic uses only
   the project's pinned TypeScript and fails fast if it isn't installed locally

**Comment by affaan-m** [comment]
This is relevant, but it is now conflict-heavy against current main: catalog counts, README inventory, package allowlist, manifests, and current frontend/a11y additions have moved since the branch was cut. I’m leaving it open, but it needs a rebase onto current main plus refreshed counts/registries before merge. Once rebased, the React track should be reviewed as its own focused feature because it touches agents, commands, rules, skills, plugin metadata, package files, and localized docs.
